### PR TITLE
Use builtin request context

### PR DESCRIPTION
**Pull request by jonbonazza**

### CODE REVIEW

File: jwtmiddleware.go [hunk 1]
```diff
@@ -222,14 +222,13 @@ func (m *JWTMiddleware) CheckJWT(w http.ResponseWriter, r *http.Request) error {
...
+		return errors.New("Token is invalid")
 	}
 
 	m.logf("JWT: %v", parsedToken)
 
 	// If we get here, everything worked and we can set the
 	// user property in context.
-	context.Set(r, m.Options.UserProperty, parsedToken)
-
+	r.WithContext(context.WithValue(r.Context(), m.Options.UserProperty, parsedToken))
```
Comment (hyfrey) [inline jwtmiddleware.go]: r.WithContext() function returns a new request, which should be passed to next http.Handler

**Comment by jonbonazza** [comment]
@hyfrey you are correct, and I actually already knew that. A bit embarrassed by that oversight. Thanks for pointing it out. I updated the PR with a quick fix. It's a sort of hacky solution in my opinion, but since we don't have a global context to use (which is the point of this change, really) this is the best I could come up with. Please re-review at your convenience.

**Comment by jgillich** [comment]
`newRequest` is not even needed since you can set `r` right away, right?

**Comment by jgillich** [comment]
Also, can any of the maintainers get this in (@ntotten)? As mentioned in #29, this is pretty important for anyone on Go 1.7+.

**Comment by jgillich** [comment]
Since this repo appears to be unmaintained, I've forked it at https://github.com/jgillich/jwt-middleware

**Comment by jgillich** [comment]
@chenkie Thanks! There are a few more issues currently though:

* Token `Claims` is now an interface
* String context keys are global, which is why they trigger a warning by golint (requires a breaking change, but necessary IMO)
* The readme needs updating

I've fixed these in my fork, but I'm unsure if I should contribute them back or just keep maintaining it myself. There is a lot of stuff here that hasn't received any attention for 1+ years.

**Comment by chenkie** [comment]
Thanks @jgillich. Can you send me an email? ryan.chenkie@auth0

**Comment by grounded042** [comment]
@jgillich My team at Auth0 is picking up maintenance of this repo - would love to chat about your thoughts on the repo and what it needs. Feel free to shoot me an email: jon.carl@auth0

**Pull request merged**